test/test.js

```diff
@@ -86,6 +86,14 @@ plantuml::test/fixtures/alice.puml[svg,role=sequence]
       const hash = rusha.createHash().update(`https://kroki.io/plantuml/svg/${encode(file)}`).digest('hex')
       expect(html).to.contain(`<img src=".asciidoctor/kroki/diag-${hash}.svg" alt="diagram">`)
     })
+    it('should convert a file containing the macro form using a relative path to a diagram', () => {
+      const registry = asciidoctor.Extensions.create()
+      asciidoctorKroki.register(registry)
+      const file = `${__dirname}/fixtures/macro/doc.adoc`
+      const html = asciidoctor.loadFile(file, { extension_registry: registry, safe: 'unsafe' }).convert()
```

Comment:
any specific reason to use `loadFile` + `convert` and not `convertFile`?

Reply:
I couldn't figure out how to use convertFile and not have to clean up the output file which didn't seem necessary.
http://asciidoctor.github.io/asciidoctor.js/master/#asciidoctor didn't seem to help.

Reply:
More specifically, the API docs suggest that the output of `convertFile` is:

> Returns
> (string | Document): the Document object if the converted String is written to a file, otherwise the converted String
> Example
> 
> var html = asciidoctor.convertFile('./document.adoc');

But I couldn't figure out how to avoid the output to file and found this a bit confusing.

Reply:
Ah I see, then we should probably improve the documentation. If you don't want to generate a file you need to set the option `to_file` to `false`.
But now that I understand the reason using `load_file` + `convert` is also fine 👌

Reply:
Oh thank you.

The options documentation has been a bit confusing for me in the Ruby API as well. I also looked at the [convert options](https://asciidoctor-docs.netlify.app/asciidoctor.js/processor/convert-options/) page in the Asciidoctor.js docs. Although it has `to_file` it's not so obvious that setting it to false means that `convertFile` does not in fact convert file... ! And it's not listed as accepting a boolean like other options.

This in turn mirrors the Ruby API which is also not so clear (see the [user manual](https://asciidoctor.org/docs/user-manual/#ruby-api-options)). Perhaps once the great docs to Antora project finishes I can do a PR for this.

Reply:
Yes, definitely!
Especially since this setting can be a string or a boolean which can in turn change the return type of the `convertFile` method.